src/Plugin/Field/FieldWidget/OgComplex.php

```diff
@@ -26,13 +31,310 @@
 class OgComplex extends EntityReferenceAutocompleteWidget {
 
   /**
-   * The OG complex widget have a special logic on order to return the groups
-   * that user can reference to.
+   * {@inheritdoc}
    */
   public function formElement(FieldItemListInterface $items, $delta, array $element, array &$form, FormStateInterface $form_state) {
     $parent = parent::formElement($items, $delta, $element, $form, $form_state);
+    // todo: fix the definition in th UI level.
     $parent['target_id']['#selection_handler'] = 'default:og';
+    $parent['target_id']['#selection_settings']['field_mode'] = 'default';
+
     return $parent;
   }
 
+  /**
+   * {@inheritdoc}
+   */
+  public function form(FieldItemListInterface $items, array &$form, FormStateInterface $form_state, $get_delta = NULL) {
+    $parent_form = parent::form($items, $form, $form_state, $get_delta);
+
+    $parent_form['other_groups'] = [];
+
+    // Adding the other groups widget.
+    if ($this->isGroupAdmin()) {
+      $parent_form['other_groups'] = $this->otherGroupsWidget($items, $form_state);
+    }
+
+    return $parent_form;
+  }
+
+  /**
+   * Special handling to create form elements for multiple values.
+   *
+   * Handles generic features for multiple fields:
+   * - number of widgets
+   * - AHAH-'add more' button
+   * - table display and drag-n-drop value reordering
+   */
+  protected function formMultipleElements(FieldItemListInterface $items, array &$form, FormStateInterface $form_state) {
+    $field_name = $this->fieldDefinition->getName();
+    $cardinality = $this->fieldDefinition->getFieldStorageDefinition()->getCardinality();
+    $parents = $form['#parents'];
+
+    $target_type = $this->fieldDefinition->getTargetEntityTypeId();
+    $user_groups = Og::getEntityGroups('user', \Drupal::currentUser()->id());
```

Comment:
The selection handler has some similar logic, we should consolidate with that too. Maybe small f/u? I'm thinking we add another helper to `Og` for that.